src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -244,6 +259,24 @@ private static Class<?> getClassType(Object base, int icol, boolean[] isScaled)
         }
     }
 
+    private static Class<?> getClassByTtype(String ttype) {
+        if (ttype==null) return String.class;
```

Comment:
This should be called ByTform and I think it will not recognize array types.

Reply:
I forgot to mention that "^[1-9]*" will not match any number with zero it them.  e.g. 10

Reply:
ok, I will clean it up.